### PR TITLE
Esther/gp 63/third party sign in

### PR DESCRIPTION
**Pull request by Esther125**
### Prerequisites
- Navigate to the Google Cloud API Credentials page at [Google Cloud Console](https://console.cloud.google.com/apis/credentials?hl=en) and create a new project.
- Follow the instructions on this page to register a client: [Google OAuth 2.0 Integration Guide](https://growingdna.com/google-oauth-2-0-for-3rd-party-login/). Please set two redirect URIs: `http://localhost:7877/google-oidc/auth` and `http://localhost:7877/google-oidc/login`.
- After registering, add yourself as a test user and download the OAuth client JSON file. Rename this file to `client.json` and place it under the `services/google_oidc` directory.

### Steps to Test
1. Update your `.env` file.
  ```
  # Google OIDC Login Settings
  GOOGLE_CLIENT_ID = [YOUR_GOOGLE_CLIENT_ID]
  GOOGLE_CLIENT_SECRET = [YOUR_GOOGLE_CLIENT_SECRET]
  GOOGLE_REDIRECT_URI = 'http://localhost:7877/google-oidc/auth'
  ```  
2.  Switch to my branch:
   ```bash
   git checkout esther/GP-63-third-party-sign-in
   ```
3.  Start the test environment:
   ```bash
   # Accessing the container's shell
   docker exec -it graduation-project /bin/bash
   uvicorn src.main:app --host localhost --port 7877 --reload

   ```
4. Visit [http://localhost:7877/google-oidc/](http://localhost:7877/google-oidc/) to see the test results.

### Expected Results
homepage

login
login-successful

### CODE REVIEW

**Review by yun-si — APPROVED** [review]
success~

File: .gitignore
Comment (frafishen) [inline .gitignore:1]: I think `client.json` should also be contained.

File: packages/google-oidc/oidc.py
Comment (frafishen) [inline packages/google-oidc/oidc.py:1]: The client of `google-oidc` should be an independent Class (just like [`PosgresqTestClient`](https://github.com/Fang-4-Group/Graduation-Project/blob/master/database/seeds/pg_api_for_testing.py)), which should be stored in the path of `./src/services/google/google_oidc.py`, and be used in the [`route.py`](https://github.com/Fang-4-Group/Graduation-Project/blob/master/src/app/route.py) to divide the functions and routing.

**Review by frafishen — CHANGES_REQUESTED** [review]
Pls also add the test for the google-oidc.

File: packages/google-oidc/oidc.py
Comment (Esther125) [reply packages/google-oidc/oidc.py:1]: Done.

**Review by frafishen — APPROVED** [review]
update the `.gitignore` and  `readme`, and it's ok to merge~

**Pull request merged**